cf/regrid/regrid.py

```diff
@@ -0,0 +1,1895 @@
+}
+
+
+@dataclass()
+class Grid:
+    """A source or destination grid definition.
+
+    .. versionadded:: TODODASKVER
+
+    """
+
+    # The domain axis identifiers of the regrid axes, in the order
+    # expected by `Data._regrid`. E.g. ['domainaxis3', 'domainaxis2']
+    axis_keys: list = field(default_factory=list)
+    # The positions of the regrid axes, in the order expected by
+    # `Data._regrid`. E.g. [3, 2]
+    axis_indices: list = field(default_factory=list)
+    # The value of the *src_axes* or *dst_axes* parameter, as
+    # appropriate.
+    axes: Any = None
+    # The sizes of the regrid axes, in the order expected by
+    # `Data._regrid`. E.g. [73, 96]
+    shape: tuple = None
+    # The regrid axis coordinates, in the order expected by
+    # `ESMF.Grid`. If the coordinates are 2-d (or more) then the axis
+    # order of each coordinate object must be as expected by
+    # `ESMF.Grid`.
+    coords: list = field(default_factory=list)
+    # The regrid axis coordinate bounds, in the order expected by
+    # `ESMF.Grid`. If the coordinates are 2-d (or more) then the axis
+    # order of each bounds object must be as expected by `ESMF.Grid`.
+    bounds: list = field(default_factory=list)
+    # For spherical regridding, whether or not the longitude axis is
+    # cyclic.
+    cyclic: Any = None
+    # The coordinate system of the grid.
+    coord_sys: str = ""
+    # The regridding method.
+    method: str = ""
+    # Identify the grid as 'source' or 'destination'.
+    name: str = ""
+    # If True then, for 1-d regridding, the ESMF weights are generated
+    # for a 2-d grid for which one of the dimensions is a size 2 dummy
+    # dimension.
+    dummy_size_2_dimension: bool = False
+
+
+def regrid(
+    coord_sys,
+    src,
+    dst,
+    method=None,
+    src_cyclic=None,
+    dst_cyclic=None,
+    use_src_mask=True,
+    use_dst_mask=False,
+    src_axes=None,
+    dst_axes=None,
+    axes=None,
+    ignore_degenerate=True,
+    return_operator=False,
+    check_coordinates=False,
+    min_weight=None,
+    inplace=False,
+    return_ESMF_regrid_operator=False,
+):
+    """Regrid a field to a new spherical or Cartesian grid.
+
+    This is a worker function primarily intended to be called by
+    `cf.Field.regridc` and `cf.Field.regrids`.
+
+    .. versionadded:: TODODASKVER
+
+    .. seealso:: `cf.Field.regridc`, `cf.Field.regrids`,
+                 `cf.data.dask_regrid.regrid`.
+
+    :Parameters:
+
+        coord_sys: `str`
+            The name of the coordinate system of the source and
+            destination grids. Either ``'spherical'`` or
+            ``'Cartesian'``.
+
+        src: `Field`
+            The source field to be regridded.
+
+        dst: `Field`, `Domain`, `RegridOperator` or sequence of `Coordinate`
+            The definition of the destination grid.
+
+            See `cf.Field.regrids` (for spherical regridding) or
+            `cf.Field.regridc` (for Cartesian regridding) for details.
```

Comment:
Since we state this for most of the keywords, could we maybe move it to the top and state it only once, instead?

Reply:
Hmm. Could do something like that, but it could get missed when only consulting a particular keyword, and it's only a worker function that we don't expect users to interact with.

Reply:
Fair enough. Happy to leave it as-is.